platform/mellanox/mlnx-platform-api/sonic_platform/module.py

```diff
@@ -247,3 +249,196 @@ def get_sfp(self, index):
         with self.lock:
             self.initialize_single_sfp(index)
             return super(Module, self).get_sfp(index)
+
+
+class DpuModule(ModuleBase):
+    config_db = ConfigDBConnector()
+    config_db.connect()
+
+    def __init__(self, dpu_id):
+        super(DpuModule, self).__init__()
+        self.dpu_id = dpu_id
+        self.dpuctl_obj = DpuCtlPlat(dpu_id - 1)
+        self.fault_state = False
+        self.vpd_parser = VpdParser('/var/run/hw-management/eeprom/vpd_data', True, self.dpu_id)
+
+    def get_base_mac(self):
+        """
+        Retrieves the base MAC address for the module
+
+        Returns:
+            A string containing the MAC address in the format
+            'XX:XX:XX:XX:XX:XX'
+        """
+        return self.vpd_parser.get_dpu_base_mac()
+
+    def reboot(self, reboot_type):
+        """
+        Request to reboot the module
+
+        Args:
+            reboot_type: A string, the type of reboot requested from one of the
+            predefined reboot types: MODULE_REBOOT_DEFAULT, MODULE_REBOOT_CPU_COMPLEX,
+            or MODULE_REBOOT_FPGA_COMPLEX
+
+        Returns:
+            bool: True if the request has been issued successfully, False if not
+        """
+        # TODO: To change dpuctl implementation to return True or False based on Success
+        if reboot_type == ModuleBase.MODULE_REBOOT_DEFAULT:
+            return self.dpuctl_obj.dpu_reboot()
+        return False
+
+    def set_admin_state(self, up):
+        """
+        Request to keep the card in administratively up/down state.
+        The down state will power down the module and the status should show
+        MODULE_STATUS_OFFLINE.
+        The up state will take the module to MODULE_STATUS_FAULT or
+        MODULE_STATUS_ONLINE states.
+
+        Args:
+            up: A boolean, True to set the admin-state to UP. False to set the
+            admin-state to DOWN.
+
+        Returns:
+            bool: True if the request has been issued successfully, False if not
+        """
+        # TODO: To change dpuctl implementation to return True or False based on Success
+        if up:
+            self.fault_state = not self.dpuctl_obj.dpu_power_on()
```

Comment:
You should check the return code and force the power on or power off if needed

Reply:
fault_state should reported only if force didn't work

Reply:
Power on internally calls force power on if it has failed, power off uses going down function to check for success, which calls force power off if it has failed. Is it required to add explicit force power on and off calls?